diagnostics/collect-wsl-logs.ps1

```diff
@@ -131,68 +131,73 @@ if ($Dump)
-$stdErrLog = "$folder/wsl_network_stderr.log"
+# Using a try/catch for some commands below, as some of them do not exist on all OS versions
 
-# Using a try/catch as some of the commands below do not exist on some OS versions
 try
 {
-    Write-Output "Get-NetAdapter" | Out-File -FilePath "$folder/Get-NetAdapter.log" -Append
     Get-NetAdapter -includeHidden | select Name,ifIndex,NetLuid,InterfaceGuid,Status,MacAddress,MtuSize,InterfaceType,Hidden,HardwareInterface,ConnectorPresent,MediaType,PhysicalMediaType | Out-File -FilePath "$folder/Get-NetAdapter.log" -Append
+}
+catch {}
 
-    Write-Output "Get-NetFirewallHyperVVMCreator" | Out-File -FilePath "$folder/Get-NetFirewallHyperVVMCreator.log" -Append
+try
+{
     Get-NetFirewallHyperVVMCreator | Out-File -FilePath "$folder/Get-NetFirewallHyperVVMCreator.log" -Append
+}
+catch {}
 
-    Write-Output "Get-NetFirewallHyperVVMSetting -PolicyStore ActiveStore" | Out-File -FilePath "$folder/Get-NetFirewallHyperVVMSetting.log" -Append
-    Get-NetFirewallHyperVVMSetting -PolicyStore ActiveStore | Out-File -FilePath "$folder/Get-NetFirewallHyperVVMSetting.log" -Append
+try
+{
+    Get-NetFirewallHyperVVMSetting -PolicyStore ActiveStore | Out-File -FilePath "$folder/Get-NetFirewallHyperVVMSetting_ActiveStore.log" -Append
+}
+catch {}
 
-    Write-Output "Get-NetFirewallHyperVProfile -PolicyStore ActiveStore" | Out-File -FilePath "$folder/Get-NetFirewallHyperVProfile.log" -Append
-    Get-NetFirewallHyperVProfile -PolicyStore ActiveStore | Out-File -FilePath "$folder/Get-NetFirewallHyperVProfile.log" -Append
+try
+{
+    Get-NetFirewallHyperVProfile -PolicyStore ActiveStore | Out-File -FilePath "$folder/Get-NetFirewallHyperVProfile_ActiveStore.log" -Append
+}
+catch {}
 
-    Write-Output "Get-NetFirewallHyperVPort" | Out-File -FilePath "$folder/Get-NetFirewallHyperVPort.log" -Append
+try
+{
     Get-NetFirewallHyperVPort | Out-File -FilePath "$folder/Get-NetFirewallHyperVPort.log" -Append
+}
+catch {}
 
-    Write-Output "hnsdiag.exe list all" | Out-File -FilePath "$folder/hnsdiag_list_all.log" -Append
-    Start-Process -FilePath "hnsdiag.exe" -ArgumentList "list all" -RedirectStandardOutput $stdOutLog -RedirectStandardError $stdErrLog -NoNewWindow -wait
-    Get-Content $stdOutLog, $stdErrLog | Out-File -FilePath "$folder/hnsdiag_list_all.log" -Append
+try
+{
+    $cmd = "hnsdiag.exe list all 2>&1 > " + "$folder/hnsdiag_list_all.log"
+    $cmd | cmd | Out-Null
+}
+catch {}
 
-    Write-Output "hnsdiag.exe list endpoints -df" | Out-File -FilePath "$folder/hnsdiag_list_endpoints.log" -Append
-    Start-Process -FilePath "hnsdiag.exe" -ArgumentList "list endpoints -df" -RedirectStandardOutput $stdOutLog -RedirectStandardError $stdErrLog -NoNewWindow -wait
-    Get-Content $stdOutLog, $stdErrLog | Out-File -FilePath "$folder/hnsdiag_list_endpoints.log" -Append
+try
+{
+    $cmd = "hnsdiag.exe list endpoints -df 2>&1 > " + "$folder/hnsdiag_list_endpoints.log"
```

Comment:
What's the reason for taking this approach instead of  let's say:

```
& hnsdiag.exe list endpoints -df 2>&1 > $folder/hnsdiag_list_endpoints.log
```

?

Reply:
I was not able to get the right syntax for the vfpctrl commands below (which need to be dynamically created) and use this approach instead

I updated for the hnsdiag commands

let me know if you think we can simplify the vfpctrl commands as well